libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -2642,50 +2701,61 @@ instance BaseUniverse fn => Functions (SetFn fn) fn where
 
 -- List -------------------------------------------------------------------
 
-data ListSpec fn a = ListSpec [a] (Spec fn Integer) (Spec fn a) (FoldSpec fn a)
+data ListSpec fn a = ListSpec (Maybe Integer) [a] (Spec fn Integer) (Spec fn a) (FoldSpec fn a)
```

Comment:
(Maybe Integer)  An upper bound on the size (if (Just n)), why an upper bound?
What is the realtionship between the first and third argument to ListSpec?

Reply:
The first is the hint for controlling test case distributio, it's not checked. Whereas the third is what is derived from checked constraints.

There is no particular reason why it had to be a maximum, one could also try to have it be an average. However, implementing average would be _super difficult_ because you don't know the constraints on size and sums etc.